### ISSUE TITLE
Missing or unaccessiable images can cause clients freeze

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#4427]
Need to check some errors with missing images -- some users report about "crash" while gameplay or re-connection.

Logs example -- error is not suppressed and go to client:

```
INFO  2018-01-18 20:54:53,488 Game 43283eb5-ed03-4a5c-b4af-ee82593d83fb started for player 9d248c78-1da4-40c2-be62-0c0947e551cf =>[AWT-EventQueue-0] CallbackClientImpl.gameStarted 
ERROR 2018-01-18 20:54:59,003 java.io.EOFException                                                                       =>[Util3] ImageCache.makeThumbnail 
java.io.EOFException
	at net.java.truecommons.io.PowerBuffer.load(PowerBuffer.java:144)
	at net.java.truevfs.comp.zip.AbstractZipFile.checkZipFileSignature(AbstractZipFile.java:162)
	at net.java.truevfs.comp.zip.AbstractZipFile.<init>(AbstractZipFile.java:137)
	at net.java.truevfs.comp.zipdriver.ZipInputService.<init>(ZipInputService.java:40)
	at net.java.truevfs.comp.zipdriver.AbstractZipDriver.newZipInput(AbstractZipDriver.java:337)
	at net.java.truevfs.comp.zipdriver.AbstractZipDriver.newInput(AbstractZipDriver.java:319)
	at net.java.truevfs.comp.zipdriver.AbstractZipDriver.newInput(AbstractZipDriver.java:46)
	at net.java.truevfs.kernel.spec.FsArchiveDriver.newInput(FsArchiveDriver.java:187)
	at net.java.truevfs.kernel.impl.TargetArchiveController.liftedTree1$1(TargetArchiveController.scala:150)
	at net.java.truevfs.kernel.impl.TargetArchiveController.mount0(TargetArchiveController.scala:149)
	at net.java.truevfs.kernel.impl.TargetArchiveController.mount(TargetArchiveController.scala:109)
	at net.java.truevfs.kernel.impl.FileSystemArchiveController$ResetFileSystem.autoMount(FileSystemArchiveController.scala:70)
	at net.java.truevfs.kernel.impl.FileSystemArchiveController.autoMount(FileSystemArchiveController.scala:37)
	at net.java.truevfs.kernel.impl.BasicArchiveController.checkAccess(BasicArchiveController.scala:56)
	at net.java.truevfs.kernel.impl.DefaultManager$BackController.net$java$truevfs$kernel$impl$SyncController$$super$checkAccess(DefaultManager.scala:158)
	at net.java.truevfs.kernel.impl.SyncController$$anonfun$checkAccess$1.apply$mcV$sp(SyncController.scala:47)
	at net.java.truevfs.kernel.impl.SyncController$$anonfun$checkAccess$1.apply(SyncController.scala:47)
	at net.java.truevfs.kernel.impl.SyncController$$anonfun$checkAccess$1.apply(SyncController.scala:47)
	at net.java.truevfs.kernel.impl.SyncController$class.net$java$truevfs$kernel$impl$SyncController$$apply(SyncController.scala:129)
	at net.java.truevfs.kernel.impl.SyncController$class.checkAccess(SyncController.scala:47)
	at net.java.truevfs.kernel.impl.DefaultManager$BackController.net$java$truevfs$kernel$impl$LockController$$super$checkAccess(DefaultManager.scala:158)
	at net.java.truevfs.kernel.impl.LockController$$anonfun$checkAccess$1.apply$mcV$sp(LockController.scala:48)
	at net.java.truevfs.kernel.impl.LockController$$anonfun$checkAccess$1.apply(LockController.scala:48)
	at net.java.truevfs.kernel.impl.LockController$$anonfun$checkAccess$1.apply(LockController.scala:48)
	at net.java.truevfs.kernel.impl.LockingStrategy.apply(LockingStrategy.scala:84)
	at net.java.truevfs.kernel.impl.LockController$class.timedReadOrWriteLocked(LockController.scala:104)
	at net.java.truevfs.kernel.impl.LockController$class.checkAccess(LockController.scala:48)
	at net.java.truevfs.kernel.impl.DefaultManager$BackController.checkAccess(DefaultManager.scala:158)
	at net.java.truevfs.kernel.impl.ArchiveControllerAdapter.checkAccess(ArchiveControllerAdapter.scala:23)
	at net.java.truevfs.comp.zipdriver.AbstractKeyController.checkAccess(AbstractKeyController.java:105)
	at net.java.truevfs.kernel.spec.FsDecoratingController.checkAccess(FsDecoratingController.java:53)
	at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$$anonfun$checkAccess$1.apply(FalsePositiveArchiveController.scala:74)
	at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$$anonfun$checkAccess$1.apply(FalsePositiveArchiveController.scala:74)
	at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$TryChild$.apply(FalsePositiveArchiveController.scala:195)
	at net.java.truevfs.kernel.impl.FalsePositiveArchiveController.net$java$truevfs$kernel$impl$FalsePositiveArchiveController$$apply(FalsePositiveArchiveController.scala:172)
	at net.java.truevfs.kernel.impl.FalsePositiveArchiveController.checkAccess(FalsePositiveArchiveController.scala:74)
	at net.java.truevfs.access.TFile.exists(TFile.java:1670)
	at org.mage.plugins.card.images.ImageCache$1.apply(ImageCache.java:119)
	at org.mage.plugins.card.images.ImageCache$1.apply(ImageCache.java:57)
	at com.google.common.collect.MapMaker$StrategyImpl.compute(MapMaker.java:592)
	at com.google.common.collect.MapMaker$StrategyImpl.compute(MapMaker.java:462)
	at com.google.common.collect.CustomConcurrentHashMap$ComputingImpl.get(CustomConcurrentHashMap.java:2045)
	at org.mage.plugins.card.images.ImageCache.getImage(ImageCache.java:405)
	at org.mage.plugins.card.images.ImageCache.getThumbnail(ImageCache.java:381)
	at org.mage.card.arcane.CardPanelRenderImpl.lambda$updateArtImage$2(CardPanelRenderImpl.java:347)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
	Suppressed: java.nio.file.NoSuchFileException: C:\Users\Owner\Desktop\xmage\mage-client\plugins\images\BNG.thumb.zip\BNG\Brimaz, King of Oreskos.full.jpg
		at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
		at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
		at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
		at sun.nio.fs.WindowsFileSystemProvider.checkReadAccess(WindowsFileSystemProvider.java:338)
		at sun.nio.fs.WindowsFileSystemProvider.checkAccess(WindowsFileSystemProvider.java:362)
		at net.java.truevfs.driver.file.FileController.checkAccess(FileController.java:101)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$$anonfun$checkAccess$1.apply(FalsePositiveArchiveController.scala:74)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$$anonfun$checkAccess$1.apply(FalsePositiveArchiveController.scala:74)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$UseParent.apply(FalsePositiveArchiveController.scala:204)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController.net$java$truevfs$kernel$impl$FalsePositiveArchiveController$$apply(FalsePositiveArchiveController.scala:178)
		... 15 more
	Suppressed: java.nio.file.NoSuchFileException: C:\Users\Owner\Desktop\xmage\mage-client\plugins\images\BNG.thumb.zip\BNG
		at sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:79)
		at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:97)
		at sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:102)
		at sun.nio.fs.WindowsFileSystemProvider.createDirectory(WindowsFileSystemProvider.java:504)
		at java.nio.file.Files.createDirectory(Files.java:674)
		at java.nio.file.Files.createAndCheckIsDirectory(Files.java:781)
		at java.nio.file.Files.createDirectories(Files.java:767)
		at net.java.truevfs.driver.file.FileOutputSocket.begin(FileOutputSocket.java:112)
		at net.java.truevfs.driver.file.FileOutputSocket.stream(FileOutputSocket.java:233)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$Output$1$$anonfun$stream$2.apply(FalsePositiveArchiveController.scala:119)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$Output$1$$anonfun$stream$2.apply(FalsePositiveArchiveController.scala:119)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$UseParent.apply(FalsePositiveArchiveController.scala:204)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController.net$java$truevfs$kernel$impl$FalsePositiveArchiveController$$apply(FalsePositiveArchiveController.scala:172)
		at net.java.truevfs.kernel.impl.FalsePositiveArchiveController$Output$1.stream(FalsePositiveArchiveController.scala:119)
		at net.java.truevfs.access.TFileOutputStream.<init>(TFileOutputStream.java:164)
		at net.java.truevfs.access.TFileOutputStream.<init>(TFileOutputStream.java:134)
		at net.java.truevfs.access.TFileOutputStream.<init>(TFileOutputStream.java:120)
		at org.mage.plugins.card.images.ImageCache.makeThumbnail(ImageCache.java:516)
		at org.mage.plugins.card.images.ImageCache$1.makeThumbnailByFile(ImageCache.java:177)
		at org.mage.plugins.card.images.ImageCache$1.apply(ImageCache.java:140)
		... 12 more
```

**Comment by Zerrisx** [comment]
Is this still an issue?

**Comment by JayDi85** [comment]
Yes, it's still actual bug with broken images files from some sets (zero size zip files). But I'm not sure about freezes or other consequences.

Solution: clean up it manually from images folder or implement auto-clean up feature (see #6267).

**Comment by JayDi85** [comment]
After image files clean up implemented (#6267) -- that's issue is no more actual.
